auth_token/models.py

```diff
@@ -40,16 +42,22 @@ def _hash_key(self, key, **kwargs):
         return hash_key(key)
 
     def create(self, key_generator, **kwargs):
-        for attempt in range(settings.MAX_RANDOM_KEY_ITERATIONS + 1):
+        for attempt in range(1, settings.MAX_RANDOM_KEY_ITERATIONS + 1):
             try:
                 key = key_generator()
                 hashed_key = self._hash_key(key, **kwargs)
                 obj = super().create(key=hashed_key, **kwargs)
                 obj.secret_key = key
                 return obj
             except IntegrityError:
```

Comment:
there can be two exceptions PersistenceException if key already exists before the save or IntegrityError if key was created into database before chamber validations and DB insert

Reply:
ouch...

Reply:
yes this was my big mistake, with missing test. Sorry for it